Dockerfile

```diff
@@ -1,10 +1,23 @@
-FROM node:lts-slim
+FROM node:16.15.1-alpine as base
+
+RUN apk --update --no-cache add --virtual .builds-deps build-base python3
+
 WORKDIR /opt/app/
-COPY package.json ./
-RUN npm install
-RUN npm prune
-COPY . /opt/app
+
+FROM base as build
+
+COPY package.json package-lock.json ./
+
+RUN npm ci
+
+COPY . .
+
+FROM base
+
+COPY --from=build /opt/app /opt/app/
+
 ARG NODE_ENV
+
 ENV NODE_ENV $NODE_ENV
 
-CMD [ "npm", "start" ]
+CMD [ "node", "app.js" ]
```

Comment:
Enable to communicate directly with the node process, enables you to handle any kind of process signalling and have a graceful shutdown. Cutting off the npm additional layer that we don't need in production, since the scripts are more for dev usage. We've being using like this in our production services